### PR TITLE
refactor website to use v3 api + dynamic checksums

### PR DESCRIPTION
**Pull request by gdams**
- Refactor website to use v3
- Implement a bootstrap popup for checksums
- Refactor Archive page

### CODE REVIEW

**Comment by karianna** [comment]
@johnoliver - Is the v3 API ready for prime time?  I wasn't sure f @smlambert and co had had a chance to swap over to it yet.

**Comment by gdams** [comment]
@karianna these API routes are dedicated to the website and very unlikely to change. I think that we should implement this now to be able to take advantage of the dynamic checksums.

**Comment by johnoliver** [comment]
I think my main thing here would be if we are going to switch over the website to v3 I would probably like to have a staging v3 instance in openshift to do test deployments, as we are still working through making it stable atm

**Comment by johnoliver** [comment]
On the staging server I am getting `Cannot read property 'link' of undefined` are you sure you took account of the fact not all platforms have an installer?

**Comment by gdams** [comment]
@johnoliver may be a cache thing? works fine for me

**Comment by johnoliver** [comment]
@karianna care to cast a tie breaker if https://staging.adoptopenjdk.net/617/ works for you?

**Comment by karianna** [comment]
@johnoliver & @gdams - got as far as https://staging.adoptopenjdk.net/617/releases.html?variant=openjdk9&jvmVariant=hotspot shows the wrong latest versions for 9 compared to live, suspect a version parsing bug

**Comment by johnoliver** [comment]
@karianna should be fixed by https://github.com/AdoptOpenJDK/openjdk-api-v3/pull/36

**Comment by karianna** [comment]
Restarted staging - still get wrong 9 version? https://staging.adoptopenjdk.net/617/releases.html?variant=openjdk9&jvmVariant=hotspot

**Comment by gdams** [comment]
@karianna that link is returning the `jdk-9.0.4+11` for me?

**Comment by gdams** [comment]
it does seem to be failing to show the jdk9-openj9 and jdk10-openj9 binaries though... @johnoliver any idea why the API isn't returning those versions?

**Comment by johnoliver** [comment]
@gdams should be fixed by https://github.com/AdoptOpenJDK/openjdk-api-v3/pull/37

**Comment by johnoliver** [comment]
looks like it is working now with the exception that it does not seemed to have pulled checksums for those builds, possibly because the binarys in those checksum files does not match the actual binary name

**Comment by gdams** [comment]
hmmm It actually appears that the checksum files have been deleted, this is weird....

**Comment by gdams** [comment]
okay so I have found the issue....

the old j9 checksum files appear to be called something like this:

`OpenJDK10-OPENJ9_ppc64le_Linux_jdk-10.0.2.13_openj9-0.9.0.sha256.txt`

whereas the newer files are called:

`OpenJDK11U-jdk_ppc64le_linux_openj9_11.0.5_10_openj9-0.17.0.tar.gz.sha256.txt`

@johnoliver can you add some logic that tries with and without the `.tar.gz` in the checksum filename?

**Comment by gdams** [comment]
I assume this requires a change to https://github.com/AdoptOpenJDK/openjdk-api-v3/blob/22ffe140ffaf52241b21f67c28206d70b76eff1a/adoptopenjdk-api-v3-updater/src/main/kotlin/net/adoptopenjdk/api/v3/mapping/adopt/AdoptBinaryMapper.kt#L111-L118

**Comment by gdams** [comment]
I think that this is good to go now

**Comment by gdams** [comment]
actually @johnoliver the order of releases by date seems wrong here: https://staging.adoptopenjdk.net/617/archive.html?variant=openjdk13&jvmVariant=openj9

**Comment by karianna** [comment]
Also https://staging.adoptopenjdk.net/617/archive.html?variant=openjdk9&jvmVariant=hotspot - Mac OS X 9.0.4 release isn't there, but we may not have actually built one.

**Comment by gdams** [comment]
looks like we didn't build one: https://github.com/AdoptOpenJDK/openjdk9-binaries/releases/tag/jdk-9.0.4%2B11

**Comment by karianna** [comment]
OK cool, so it's just the date order to go I think, I've eyeballed the rest and it seems good.  Hmm, actually https://staging.adoptopenjdk.net/617/archive.html?variant=openjdk13&jvmVariant=openj9 seems to also have a column width issue.  @gdams that might be yours to fix.

**Comment by karianna** [comment]
> Also https://staging.adoptopenjdk.net/617/archive.html?variant=openjdk9&jvmVariant=hotspot - Mac OS X 9.0.4 release isn't there, but we may not have actually built one.

OK, that's my bad, ignore that comment

**Comment by johnoliver** [comment]
@gdams @karianna I think that sorting may be done by the page, the api itself returns them in the correct order

**Comment by johnoliver** [comment]
@gdams looks like the page uses `sortByProperty` to set its own sort order

**Comment by gdams** [comment]
good spot @johnoliver. Removed

**Comment by gdams** [comment]
release order is looking much better now

**Comment by gdams** [comment]
@jerboaa check out https://staging.adoptopenjdk.net/617/upstream.html?variant=openjdk11&ga=ga and let me know what you think.

**Comment by jerboaa** [comment]
@gdams Nice work! I see you've removed the `archive` and `nightly` buttons. What's missing from the current page are source downloads, though. Unless I'm missing something.

**Comment by gdams** [comment]
Yeah I can add the source download, just need to find a suitable place to add the link. Any ideas?

**Comment by jerboaa** [comment]
> Yeah I can add the source download, just need to find a suitable place to add the link. Any ideas?

Perhaps as another "arch" in the dropdown. Failing that as a link in the section where JDK/JRE links are being placed (i.e. make it a triplet over a duet). No better ideas than that, sorry.

**Comment by gdams** [comment]
looks like bootstrap has messed up the sidebar. I'll need to fix that

**Comment by jerboaa** [comment]
@gdams Looks good. One nit left with the source link css is that the "visited" link font color is the same as the background, so once visited the link seems to "disappear" :)

**Comment by karianna** [comment]
@gdams - merge conflict.

**Comment by gdams** [comment]
@jerboaa this should be good to go now, please review and we can get this merged!

File: src/handlebars/partials/modal.handlebars [hunk 1]
```diff
@@ -0,0 +1,22 @@
+      <!-- The Modal -->
+      <div class="modal fade" id="myModal">
+        <div class="modal-dialog modal-lg">
+            <div class="modal-content">
+
+                <!-- Modal Header -->
+                <div class="modal-header">
+                <h4 class="modal-title">Checksum</h4>
```
Comment (karianna) [inline src/handlebars/partials/modal.handlebars]: Nitpick could tab this in whitespace

File: src/handlebars/partials/modal.handlebars [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+                </div>
+
+                <!-- Modal body -->
+                <div class="modal-body" id="modal-body">
+                </div>
+
+                <!-- Modal footer -->
+                <div class="modal-footer">
+                <button type="button" class="blue-button" onclick="hideChecksum()">Close</button>
```
Comment (karianna) [inline src/handlebars/partials/modal.handlebars]: whitespace indent

File: src/handlebars/upstream.handlebars [hunk 3]
```diff
@@ -1,236 +1,112 @@
...
+          </script>
+         </div>
+      </div>
+    </div>
+</div>
 
 </main>
 
-{{> footer }}
+{{> footer script='<script src="https://cdnjs.cloudflare.com/ajax/libs/moment.js/2.22.2/moment.min.js"></script>' }}
```
Comment (karianna) [inline src/handlebars/upstream.handlebars]: We seem to repeat this on a few pages - should we make it common?

File: src/js/common.js [hunk 4]
```diff
@@ -140,22 +145,32 @@ function queryAPI(release, url, openjdkImp, type, errorHandler, handleResponse)
...
+    releaseType = 'ga'
+  } else if (releaseType == 'nightly') {
+    releaseType = 'ea'
+  }
+
+  let url = `https://api.adoptopenjdk.net/v3/assets/feature_releases/${variant.replace(/\D/g,'')}/${releaseType}`
 
-  const url = `https://api.adoptopenjdk.net/v2/info/${releaseType}/${variant}`;
-  queryAPI(release, url, openjdkImp, type, errorHandler, handleResponse);
+  if (releaseType == 'ea') {
```
Comment (karianna) [inline src/js/common.js]: should be  `===`?

File: src/js/upstream.js [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+      ga: gaButton.value
+    });
+};
+
+for (let button of gaButtons) {
+    if (button.value === ga) {
+        button.setAttribute('checked', 'checked');
+        break;
+    }
+  }
```
Comment (karianna) [inline src/js/upstream.js]: whitespace

**Pull request merged**